core-processor/src/executor.rs

```diff
@@ -368,12 +375,12 @@ where
                 match status {
                     Status::Normal => (),
                     Status::GasLimitExceeded => {
-                        termination = TerminationReason::Trap(TrapExplanation::Core(
+                        termination = ActorTerminationReason::Trap(TrapExplanation::Ext(
                             ExtError::Execution(gear_core_errors::ExecutionError::GasLimitExceeded),
-                        ))
+                        ));
                     }
                     Status::GasAllowanceExceeded => {
-                        termination = TerminationReason::GasAllowanceExceeded
+                        termination = ActorTerminationReason::GasAllowanceExceeded;
```

Comment:
GasAllowanceExceeded seems not to be the actor reason at all, isn't it?

Reply:
It's anyway matched into `DispatchResultKind`

Reply:
todo to remove from Actor Reason